rfcs/49-animation-primitives.md

```diff
@@ -0,0 +1,154 @@
+# Feature Name: `animation-primitives`
+
+## Summary
+
+Animation is a particularly complex, with many stateful and intersecting
+elements. This RFC aims to detail a set of lowest-level APIs for authoring and
+playing animations within Bevy.
+
+## Motivation
+
+Animation is at the heart of modern game development. A game engine without an
+animation system is generally considered not yet production ready.
+
+This RFC aims to detail the absolute lowest level APIs to unblock ecosystem
+level experimentation with building more complex animation systems (i.e. inverse
+kinematics, animation state machine, humanoid retargetting, etc.)
+
+## Scope
+
+Animation is a huge area that spans multiple problem domains:
+
+ 1. **Storage**: this generally covers on-disk storage in the form of assets as
+    well as the in-memory representation of static animation data.
+ 2. **Sampling**: this is how we sample the animation assets over time and
+    transform it into what is applied to the animated entities.
+ 3. **Application**: this is how we applied the sampled values to animated
+    entities.
+ 4. **Composition**: this is how we compose simple clips to make more complex
+    animated behaviors.
+ 4. **Authoring**: this is how we create the animation assets used by the engine.
+
+This RFC specifically aims to resolve only problems within the domains of storage
+and sampling. Application can be distinctly decoupled from these earlier two
+stages, treating the sampled values as a black box output, and composition and
+authoring can be built separately upon the primitives provided by this RFC and
+thus are explicit non-goals here.
+
+## User-facing explanation
```

Comment:
IMO this deserves an explanation of how "animation is anything that changes over time in a smooth and predictable way".

A lot of new users, especially those who haven't been exposed to animation before, have a conceptualization of animation that only includes modifying the `Transform` of entities.

Reply:
This isn't entirely true, since there are discrete valued animations as well, so the wording here might need a bit more wordsmithing.